File: src/problem22.py
def stripQuotes(str):
    return str.strip('"')

def nameScore(s,i):
    total = 0
    for c in s:
        total += ord(c) - 64
    return total * i

def problem22():
    f = open('../data/p22_names.txt', 'r')
    nameList = map(stripQuotes,f.read().split(","))
    nameList.sort()
    total = 0
    for i,name in enumerate(nameList):
        total += nameScore(name,i+1)
    return total

print(problem22())

File: src/problem22.hs
import Data.Char (ord)
import Data.List

main = do
    contents <- readFile "../data/p22_names.txt"
    putStrLn $ show $ answer contents

answer :: String -> Int
answer xs = sum $ map nameSum $ zip [1..length xs] $ sort $ read $ "[" ++ xs ++ "]"

nameSum :: (Int,String) -> Int
nameSum (i,str) = i * (sum $ map (64 `subtract`) $ map ord str)
